File: sample-tests/typescript/helpers/sdk-fixture.ts
import { CardinalSdk } from "@icure/cardinal-sdk"

let cachedSdk: CardinalSdk | undefined

/**
 * Returns an initialized CardinalSdk instance, cached for reuse across tests.
 * Reads credentials from environment variables.
 *
 * Required env vars:
 *   CARDINAL_URL       — e.g. "https://api.icure.cloud"
 *   CARDINAL_USERNAME  — login username
 *   CARDINAL_PASSWORD  — login password
 */
export async function getTestSdk(): Promise<CardinalSdk> {
  if (cachedSdk) return cachedSdk

  const url = process.env.CARDINAL_URL
  const username = process.env.CARDINAL_USERNAME
  const password = process.env.CARDINAL_PASSWORD

  if (!url || !username || !password) {
    throw new Error(
      'Missing required environment variables: CARDINAL_URL, CARDINAL_USERNAME, CARDINAL_PASSWORD'
    )
  }

  // TODO: Replace with actual CardinalSdk.initialize() call.
  // The exact initialization API depends on the SDK version.
  // Example:
  //   cachedSdk = await CardinalSdk.initialize(url, username, password)
  //   return cachedSdk
  throw new Error('SDK initialization not yet implemented — update helpers/sdk-fixture.ts')
}

File: sample-tests/kotlin/src/test/kotlin/com/icure/docs/helpers/SdkFixture.kt
package com.icure.docs.helpers

import com.icure.cardinal.sdk.CardinalSdk

object SdkFixture {
    private var cachedSdk: CardinalSdk? = null

    /**
     * Returns an initialized CardinalSdk instance, cached for reuse across tests.
     * Reads credentials from environment variables.
     *
     * Required env vars:
     *   CARDINAL_URL       - e.g. "https://api.icure.cloud"
     *   CARDINAL_USERNAME  - login username
     *   CARDINAL_PASSWORD  - login password
     */
    suspend fun getTestSdk(): CardinalSdk {
        cachedSdk?.let { return it }

        val url = System.getenv("CARDINAL_URL")
            ?: error("Missing required environment variable: CARDINAL_URL")
        val username = System.getenv("CARDINAL_USERNAME")
            ?: error("Missing required environment variable: CARDINAL_USERNAME")
        val password = System.getenv("CARDINAL_PASSWORD")
            ?: error("Missing required environment variable: CARDINAL_PASSWORD")

        // TODO: Replace with actual CardinalSdk initialization.
        // Example:
        //   cachedSdk = CardinalSdk.initialize(url, AuthenticationMethod.UsingCredentials(UsernamePassword(username, password)))
        //   return cachedSdk!!
        error("SDK initialization not yet implemented -- update SdkFixture.kt")
    }
}
